libbeat/processors/actions/cloud_metadata.go

```diff
@@ -0,0 +1,224 @@
+package actions
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+	s "github.com/elastic/beats/metricbeat/schema"
+	c "github.com/elastic/beats/metricbeat/schema/mapstriface"
+	"github.com/pkg/errors"
+)
+
+const (
+	ec2MetadataHost        = "169.254.169.254"
+	ec2InstanceIdentityURI = "/2014-02-25/dynamic/instance-identity/document"
+
+	digitaloceanMetadataHost = "169.254.169.254"
+	digitaloceanMetadataURI  = "/metadata/v1.json"
+
+	gceMetadataHost = "169.254.169.254"
+	gceMetadataURI  = "/computeMetadata/v1/?recursive=true&alt=json"
+)
+
+var (
+	metaClient = &http.Client{
+		Transport: &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout:   2 * time.Second,
+				KeepAlive: 0, // We are only making a single request.
+			}).Dial,
+			ResponseHeaderTimeout: 2 * time.Second,
```

Comment:
I chose 5 to allow the individual requests to either complete or timeout on their own. In the worst case, timeout could take ~4 seconds (2 seconds for connect timeout + 2 seconds for response header timeout). I left a 1 second buffer since it's execute three requests in parallel and they might not all be scheduled immediately.